socorro/lib/search_common.py

```diff
@@ -310,6 +321,42 @@ def fix_hang_type_parameter(parameters):
                 hang_type.value = new_values
                 hang_type.data_type = 'int'
 
+    @staticmethod
+    def fix_version_parameter(parameters):
+        """Correct the version parameter.
+
+        If version is something finishing with a 'b' and operator is
+        'has terms', replace the operator with 'starts with' to query all
+        beta versions.
+        """
+        if 'version' in parameters:
+            marked_for_deletion = []
+            for version in parameters['version']:
+                if version.operator:
+                    # We only care about the '' operator ("has terms").
+                    continue
+
+                new_values = []
+                for val in version.value:
+                    if val.endswith('b'):
+                        parameters['version'].append(SearchParam(
+                            name='version',
+                            value=[val],
+                            data_type='str',
+                            operator='$',
+                            operator_not=version.operator_not,
+                        ))
+                    else:
+                        new_values.append(val)
+
+                if new_values:
+                    version.value = new_values
+                else:
+                    marked_for_deletion.append(version)
+
+            for param in marked_for_deletion:
+                parameters['version'].remove(param)
+
```

Comment:
Also, if you loop over a copy you don't get the weird thing of adding a value to the list during the iteration.